### PR TITLE
Support unified configuration

### PR DESCRIPTION
**Pull request by rcohencyberarmor**
## PR Type:
Enhancement

___
## PR Description:
This pull request adds support for unified configuration in the codebase. It includes changes to several files and introduces new features and improvements.

___
## PR Main Files Walkthrough:
- `core/cautils/getter/kscloudapi_test.go`: Removed unused imports and commented out a test function.
- `core/mocks/loadmocks.go`: Added a new function `MockFramework_0044` to mock a framework.
- `core/cautils/sliceutils_test.go`: Added a new test function `TestIsSubSliceScanningScopeType` to test a utility function.
- `core/pkg/resourcehandler/k8sresources.go`: Updated the function `GetResources` to use the new `identifiers` package instead of `armotypes` package.
- Several other files were modified as well.

### CODE REVIEW

**Comment by dwertent** [comment]
@rcohencyberarmor please fix the conflicts

File: core/cautils/customerloader.go [hunk 1]
```diff
@@ -63,6 +67,44 @@ func (co *ConfigObj) Config() []byte {
 	return []byte{}
 }
 
+func (co *ConfigObj) UpdateEmptyFields(inCO *ConfigObj) error {
```
Comment (dwertent) [inline core/cautils/customerloader.go:70]:
```suggestion
func (co *ConfigObj) updateEmptyFields(inCO *ConfigObj) error {
```
Should be a private function

File: core/cautils/customerloader.go [hunk 2]
```diff
@@ -63,6 +67,44 @@ func (co *ConfigObj) Config() []byte {
...
+	if inCO.AccountID != "" {
+		co.AccountID = inCO.AccountID
+	}
+	if inCO.ClientID != "" {
```
Comment (dwertent) [inline core/cautils/customerloader.go:74]: ClientID and secretKey are deprecated. You can fully ignore them.

File: core/cautils/customerloader.go [hunk 3]
```diff
@@ -63,6 +67,44 @@ func (co *ConfigObj) Config() []byte {
...
+	if inCO.CustomerAdminEMail != "" {
+		co.CustomerAdminEMail = inCO.CustomerAdminEMail
+	}
+	if inCO.SecretKey != "" {
+		co.SecretKey = inCO.SecretKey
+	}
+	if inCO.Token != "" {
+		co.Token = inCO.Token
+	}
+	if inCO.CustomerGUID != "" {
```
Comment (dwertent) [inline core/cautils/customerloader.go:101]: CustomerGUID is deprecated.
You can fully ignore it.
Also, you can remove it from the config struct definition

File: core/cautils/customerloader.go [hunk 4]
```diff
@@ -554,15 +613,15 @@ func getConfigMapName() string {
 	if n := os.Getenv("KS_DEFAULT_CONFIGMAP_NAME"); n != "" {
 		return n
 	}
-	return "kubescape"
+	return kubescapeConfigMapName
```
Comment (dwertent) [inline core/cautils/customerloader.go:616]: Well done

File: core/cautils/customerloader_test.go [hunk 5]
```diff
@@ -308,7 +308,7 @@ func TestGetConfigMapNamespace(t *testing.T) {
 	}{
 		{
 			name: "no env",
-			want: "default",
+			want: "kubescape",
```
Comment (dwertent) [inline core/cautils/customerloader_test.go:311]:
```suggestion
			want: kubescapeNamespace,
```

File: core/cautils/sliceutils.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package cautils
+
+import "github.com/kubescape/opa-utils/reporthandling"
+
+func IsSubSliceScanningScopeType(haystack []reporthandling.ScanningScopeType, needle []reporthandling.ScanningScopeType) bool {
```
Comment (dwertent) [inline core/cautils/sliceutils.go:5]: What is this function?

**Review by dwertent — CHANGES_REQUESTED** [review]
Please see my inline comments

File: core/cautils/sliceutils.go (hunk 6 above)
Comment (rcohencyberarmor) [reply core/cautils/sliceutils.go:5]: I'll remove it, it is not used now.(I thought to keep it since it is already tested)

**Comment by dwertent** [comment]
The [test PR](https://github.com/armosec/system-tests/pull/140) has been merged

**Comment by dwertent** [comment]
Also, Why did you [remove this test](https://github.com/kubescape/kubescape/pull/1304/commits/504560a1965cc0a9cff595400253bfc7102ec060)?

**Comment by rcohencyberarmor** [comment]
> Also, Why did you [remove this test](https://github.com/kubescape/kubescape/pull/1304/commits/504560a1965cc0a9cff595400253bfc7102ec060)?
I see that test are passing locally but they are failing here, they are failing since the config file os the kubescape is not exist on the machine, and I think that what may cause it is the deletion test(that I removed), but it won't fix it, so I need farther investigation why it is failing just here. but it probably related to the this commit on the system test (https://github.com/armosec/system-tests/pull/149/commits) and that we don't recreate the config.json file under the ~/.kubescape when we redownload the kubescape after we delete the config.json file

**Review by dwertent — APPROVED** [review]

**Pull request merged**